manifests/master.pp

```diff
@@ -328,18 +328,32 @@
 
     $_conf_base = dirname($confdir)
 
-    file { [$_conf_base, $confdir, $codedir, $rundir, $ssldir]:
+    file { [$_conf_base, $confdir, $codedir]:
       ensure => 'directory',
       owner  => 'root',
-      group  => 'puppet',
+      group  => $facts['puppet_settings']['master']['group'],
       mode   => '0640'
     }
 
+    # Mode is managed by puppet itself
+    file { $rundir:
+      ensure => 'directory',
+      owner  => $facts['puppet_settings']['master']['user'],
+      group  => $facts['puppet_settings']['master']['group']
+    }
+
+    # Mode is managed by puppet itself
+    file { $ssldir:
+      ensure => 'directory',
+      owner  => $facts['puppet_settings']['master']['user'],
+      group  => $facts['puppet_settings']['master']['group']
```

Comment:
should just create a group and owner variable with how many times it is used, will be far easier to maintain. Or instead of defining it in manifests/master/sysconfig.pp define it in the top level and use that in all the rest.

Reply:
request changes, see previous